### DISCUSSION TITLE
PorousFlow THM poroelastic eqs

### DISCUSSION THREAD

**Discussion by AmandaLug** [idaholab/moose#17453] [Q&A Modules: General] [answered]
Hi all,

I'm modeling a THM problem in PorousFlow and two questions came to my mind during the mechanical coupling and I could not make my mind around. I'm sorry if this is too simple and not MOOSE related...

1.- In the earth_tide_fullsat.i model, FunctionDirichletBC are implemented to model strains. From the governing equations, Dirichlet BCs should be used for displacement and not strain, right?

2.- The barometric_fully_confined.i model, works fine in that simple set up. However, if gravity is implemented, is not possible to obtain the barometric efficiency of the system anymore? I'm not sure which is the relation between effective stress and density.

Thanks!
Amanda

**Comment by WilkAndy** [comment, marked as answer]
Hi @AmandaLug ,

Thanks for your questions.  They'll help future readers too!

1. You are correct, `DirichletBC` sets a condition on the displacement.  Because this particular model has only 1 element, this immediately sets the strain.  However, now imagine a real situation with complicated, heterogeneous geology.  We know that earth-tide software such as [ETERNA-x](http://ggp.bkg.bund.de/eterna/) will predict strains at all points within the earth, as a function of time.  The simplest thing to assume is that those strains are correct, on-average, throughout your complicated, heterogeneous geology.  At some points, the strains in your geology are a little larger than ETERNA-x, and at other points, the strains are a little smaller, but their mean is the same as the ETERNA-x prediction.  So, we use a suitable `DirichletBC` on the displacements to enforce the average strain throughout our model to be the same as the ETERNA-x prediction.  Given those `DirichletBC`, MOOSE will solve for the displacements (and strains) throughout our model, finding some strains are a little larger than average, some are a little smaller, but the mean strain is fixed by the BCs, to be equal to the ETERNA-x strains.  Make sense?
2. You can still compute barometric efficiency when you have gravity.  The setup is the same.  The atmospheric pressure applies a total stress to the top of your model: it applies a `FunctionNeumannBC` to the `displacement` variable, and a `FunctionDirichletBC` to your `porepressure` variable (which you would have in a realistic situation, but `barometric_fully_confined.i` does not have).  See `atm_tides.i` for an example.  Looking at Fig4 of the documentation, i estimate barometric efficiency to be around 0.4 (ratio of orange amplitude to blue amplitude).

**Reply by AmandaLug** [reply]
Thanks Andy for the great answers! I will need some time to process this.
